EIPS/eip-rich-site-proposed-contract-metadata.md

```diff
@@ -0,0 +1,100 @@
+status: Draft
+type: Standards
+category: Interface
+created: 2022-06-15
+---
+
+## Abstract
+
+Many types of phishing attack today rely on users not knowing what contract they're interacting with, even when it's a contract the user has interacted with before. To facilitate in recognizing these interactions in a way that requires no central registry or additioanl services, we propose that wallet methods that may interact with an address should also accept proposed metadata from the site suggesting the transaction, which can be used to enhance the display of future interactions with those contracts.
+
+## Motivation
+
+We've got a lot of ways we try to gather the information for confirmations, but it ranges between very complicated, unreliable, centralized, and incomplete, and is not getting the job done.
+1. Contract names
+  - Local address book
+    - Requires user interaction, which adds UX burden, and most users skip.
+  - Centralized registries
+    - Are perpetually incomplete, and favor the established, not the upstart.
+   - reverse-resolved ENS names
+     - Attackers can choose familiar-looking names, so these are not a reliable basis of secure review of what you're looking at unless you entered it yourself.
+2. Method information
+  - Registry lookups (4byte directory, the Parity on-chain registry)
+    - Can be prone to collisions (like At Inversebrah, with trolling increasing)
+    - Returns ABIs
+      - Has no parameter names, leaving most of the transaction illegible.
+  - Trusting the contract for its own metadata (EIP-719, EIP-4430)
+    - Requires an active connection to the blockchain.
+    - Is not widespread (no backwards compatibility)
+  - [Transaction Insights](https://metamask.zendesk.com/hc/en-us/articles/4412543412123-Transaction-Insight)
+    - Requires an additional fairly heavy-weight process
+    - Is largely centralized today
+    - Only works for verified contracts on etherscan (although designed for eventual Sourcify support).
+
+Additionally, this method serves to help populate the wallet with trustworthy information about what it holds. It reduces the need to rely on any central services for indexing (enhancing performance and privacy), and eliminates dangers that come from index-based automatic asset detection, like receiving harrassment, or [Airdrop scams](https://medium.com/metamask/phisher-watch-airdrop-scams-82eea95d9b2a).
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Wallets may integrate the ability to propose names for contracts, methods, and parameters into any method that touches an address, so the wallet can be expanded with this metadata at runtime. These methods include `eth_sendTransaction`, and `eth_signTypedData` (and its variants).
```

Comment:
This isn't nearly well enough specified.  Note that this isn't a blocker for merging as a draft, but is definitely a blocker for moving to Review.

Reply:
I agree.